apps/ledger-live-mobile/src/screens/Swap/Form/TxForm/From.tsx

```diff
@@ -0,0 +1,160 @@
+import { SwapFormProps } from "../../types";
+
+interface Props {
+  provider?: string;
+  swapTx: SwapTransactionType;
+  pairs: Pair[];
+}
+
+export function From({ swapTx, provider, pairs }: Props) {
+  const { t } = useTranslation();
+  const navigation = useNavigation<SwapFormProps>();
+
+  const accounts = useAccounts(pairs);
+  const { currency, name, balance, unit } = useMemo(() => {
+    const { currency, account } = swapTx.swap.from;
+
+    return {
+      account,
+      currency,
+      name: account && getAccountName(account),
+      balance:
+        (account &&
+          currency &&
+          formatCurrencyUnit(currency.units[0], account.balance, {
+            showCode: true,
+          })) ??
+        "",
+      unit: account && getAccountUnit(account),
+    };
+  }, [swapTx.swap.from]);
+
+  usePickDefaultAccount(
+    accounts,
+    swapTx.swap.from.account,
+    swapTx.setFromAccount,
+  );
+
+  const onPress = useCallback(() => {
+    // @ts-expect-error
+    navigation.navigate("SelectAccount", {
+      target: "from",
+      accounts,
+      provider,
+      currencyIds: [...new Set(pairs.map(p => p.from))],
+    });
+  }, [navigation, accounts, provider, pairs]);
+
+  return (
+    <Flex
+      borderBottomWidth={1}
+      borderColor="neutral.c70"
+      paddingBottom={4}
+      marginBottom={4}
+    >
+      <Text>{t("transfer.swap2.form.from")}</Text>
+      <Flex flexDirection="row" justifyContent="space-between">
+        <Selector
+          currency={currency}
+          title={name}
+          subTitle={balance}
+          onPress={onPress}
+        />
+
+        <Flex flex={1} justifyContent="center">
+          <AmountInput
+            value={swapTx.swap.from.amount}
+            editable={!swapTx.swap.isMaxEnabled}
+            unit={unit}
+            onChange={swapTx.setFromAmount}
+            error={swapTx.fromAmountError}
+          />
+        </Flex>
+      </Flex>
+    </Flex>
+  );
+}
+
+type AccountLikeWithFilter = AccountLike & { disabled: boolean };
+
+// based fromSelector on apps/ledger-live-desktop/src/renderer/actions/swap.js
+function useAccounts(pairs: Pair[]): AccountLikeWithFilter[] {
+  const accounts = useSelector(shallowAccountsSelector);
+
+  const filtered = useMemo<AccountLikeWithFilter[]>(() => {
+    if (pairs === null || pairs === undefined) return [];
+
+    return flattenAccounts(accounts).map(account => {
+      const id = getAccountCurrency(account).id;
+      const isAccountAvailable = !!pairs.find(pair => pair.from === id);
+      return { ...account, disabled: !isAccountAvailable };
+    });
+  }, [accounts, pairs]);
+
+  const sorted = useMemo(() => {
+    let activeAccounts: AccountLikeWithFilter[] = [];
+    let disabledAccounts: AccountLikeWithFilter[] = [];
+    let subAccounts = [];
+    let disabledSubAccounts = [];
+
+    // Traverse the accounts in reverse to check disabled accounts with active subAccounts
+    for (let i = filtered.length - 1; i >= 0; i--) {
```

Comment:
This looks a lot like one of my comments on an util from common, can't we use this util instead of rewriting here ?